site/pages/token-revokes.js

```diff
@@ -0,0 +1,359 @@
+            }
+          })
+        })
+    },
+    [
+      library,
+      localStorageKey,
+      account,
+      syncBlock,
+      setSyncBlock,
+      setSyncStatus,
+      lastBlockNumber,
+      syncStatus
+    ]
+  )
+
+  // this effect listens for new logs that take place while the user is syncing. This will make the ux smoother
+  // though they won't be added to the sync process
+  useEffect(
+    function () {
+      if (!active) {
+        return
+      }
+
+      const subscription = library.eth.subscribe('logs', {
+        topics: [
+          APPROVAL_TOPIC,
+          library.utils.padLeft(account.toLowerCase(), 64)
+        ]
+      })
+
+      subscription.on('data', function (log) {
+        setSyncBlock((prev) => ({
+          ...prev,
+          tokenApprovals: getNewestApprovals({
+            library,
+            logs: parseLogs([log]),
+            tokenApprovals: prev.tokenApprovals
+          })
+        }))
+      })
+
+      return function () {
+        subscription.unsubscribe()
+      }
+    },
+    [active, library, account, chainId, setSyncBlock]
+  )
+
+  return { ...syncBlock, syncStatus: syncStatus, setSyncStatus }
+}
+
+const useErc20Token = function (address) {
+  const { active, library } = useWeb3React()
+  return useSWR(active ? address : null, function () {
+    const erc20Service = createErc20(library, address)
+    return Promise.all([
+      erc20Service.symbol(),
+      erc20Service.decimals(),
+      erc20Service.totalSupply()
+    ])
+  })
+}
+
+const Allowance = function ({ address, data }) {
+  const { library } = useWeb3React()
+  const { t } = useTranslation('common')
+
+  const { data: token } = useErc20Token(address)
+
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [, decimals, totalSupply] = token
+  const allowanceInWei = library.utils.hexToNumberString(data)
+  const value = toFixed(fromUnit(allowanceInWei, decimals), 6)
+  const isUnlimited = Big(totalSupply).times(10).lt(allowanceInWei)
+  return <span className="m-auto">{isUnlimited ? t('unlimited') : value}</span>
+}
+
+const Token = function ({ address }) {
+  const { data: token } = useErc20Token(address)
+  if (!token) {
+    return <span className="m-auto"></span>
+  }
+  const [symbol] = token
+  return <span className="m-auto">{symbol}</span>
+}
+
+const TokenRevokes = function () {
+  const { active } = useWeb3React()
+  const { erc20 } = useContext(PureContext)
+  const { t } = useTranslation('common')
+  const [isRevoking, setIsRevoking] = useState(false)
+
+  const { tokenApprovals, syncStatus, setSyncStatus } = useTokenApprovals()
+
+  const handleRevoke = function (address, spender) {
+    if (!active) {
+      return
+    }
+    setIsRevoking(true)
+    const erc20Service = erc20(address)
+    erc20Service
+      .revoke(spender)
+      .then(() => setIsRevoking(false))
+      .catch(() => setSyncStatus(SyncStatus.Error))
+  }
+
+  return (
+    <Layout walletConnection>
```

Comment:
Let's add a title like "List and revoke token approvals" as we have on the other mini-apps.